src/com/palmergames/bukkit/towny/event/damage/TownBlockExplosionTestEvent.java

```diff
@@ -0,0 +1,65 @@
+package com.palmergames.bukkit.towny.event.damage;
+
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An event thrown when Towny will determine the explosion status of 
+ * a townblock, or plot, in a town.
+ */
+public class TownBlockExplosionTestEvent extends Event {
+
+		private static final HandlerList handlers = new HandlerList();
+		private final Town town;
+		private final TownBlock townBlock;
+		private boolean explosion;
+
+		public TownBlockExplosionTestEvent(TownBlock townBlock, Town town, boolean explosion) {
```

Comment:
Town town can be removed from the constructor and event variables.

```java
public Town getTown() {
    return getTownBlock().getTownOrNull()
}
```

Can then be used below.

Reply:
Ignore this, Warrior says it might make intellij cranky.